packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -343,6 +327,62 @@ public function isMethodStaticCallOrClassMethodObjectType(Node $node, ObjectType
         return $classReflection->isSubclassOf($objectType->getClassName());
     }
 
+    private function resolveArrayDimFetchType(
+        ArrayDimFetch $arrayDimFetch,
+        Scope $scope,
+        Type $originalNativeType
+    ): Type {
+        /**
+         * Allow pull type from
+         *
+         *      - native function
+         *      - always defined by assignment
+         *
+         * eg:
+         *
+         *  $parts = parse_url($url);
+         *  if (!empty($parts['host'])) { }
+         *
+         * or
+         *
+         *  $parts = ['host' => 'foo'];
+         *  if (!empty($parts['host'])) { }
+         */
+        $nativeVariableType = $scope->getNativeType($arrayDimFetch->var);
+        if ($nativeVariableType instanceof MixedType || ($nativeVariableType instanceof ArrayType && $nativeVariableType->getItemType() instanceof MixedType)) {
+            return $originalNativeType;
+        }
+
+        $type = $scope->getType($arrayDimFetch);
+
+        if (! $arrayDimFetch->dim instanceof String_) {
```

Comment:
This looks fishy. We will surely find array shapes with int keys

Reply:
I think that's need another fixture for next improvement 👍